### PR TITLE
Bugfix/red 266 topic grade grading page

### PR DESCRIPTION
**Pull request by red-tt-bear**

### CODE REVIEW

File: src/Extensions/ArrayExtensions.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+declare global {
+    interface Array<T> {
+        first?: T;
+    }    
+}
+
+Object.defineProperty(Array.prototype, 'first', {
```
Comment (PanopticaRising) [inline src/Extensions/ArrayExtensions.ts:7]: :joy:

File: src/Extensions/NumberExtensions.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    }    
+}
+
+/**
+ * Meant for numbers between 0 and 1 (but doesn't have to be)
+ * Multiplies by 100 and goes to 1 decimal place
+ */
+Number.prototype.toPercentString = function (): string {
+    // Number vs number discrepancy
+    return (this as number * 100).toFixed(1);
```
Comment (PanopticaRising) [inline src/Extensions/NumberExtensions.ts:13]: Should this have a `%` after it if it's a string?

**Review by PanopticaRising — APPROVED** [review]

File: src/Extensions/NumberExtensions.ts (hunk 2 above)
Comment (doombot-exe) [reply src/Extensions/NumberExtensions.ts:13]: yeah, I'm down with that

File: src/Courses/TopicGrades/GradeInfoHeader.tsx [hunk 3]
```diff
@@ -170,6 +190,9 @@ export const GradeInfoHeader: React.FC<GradeInfoHeaderProps> = ({
 
     return (
         <Grid container spacing={1} style={{ paddingLeft: '1rem' }} alignItems='flex-start'>
+            <Grid item xs={12}>
+                <h3>Topic Score: {topicGrade?.toPercentString() ?? '--'}</h3>
```
Comment (doombot-exe) [inline src/Courses/TopicGrades/GradeInfoHeader.tsx:194]: I feel like this should include the word "total" -- something should distinguish it from the other "scores" on this page...

**Review by doombot-exe — APPROVED** [review]

**Pull request merged**